frame/vamm/src/helpers/vamm.rs

```diff
@@ -11,8 +11,7 @@ impl<T: Config> Pallet<T> {
 	pub fn get_vamm_state(vamm_id: &T::VammId) -> Result<VammStateOf<T>, DispatchError> {
 		// Requested vamm must exists and be retrievable.
 		ensure!(VammMap::<T>::contains_key(vamm_id), Error::<T>::VammDoesNotExist);
-		let vamm_state = VammMap::<T>::get(vamm_id).ok_or(Error::<T>::FailToRetrieveVamm)?;
-		Ok(vamm_state)
+		VammMap::<T>::get(vamm_id).ok_or_else(|| Error::<T>::FailToRetrieveVamm.into())
```

Comment:
Can't these be merged into one? Basically by keeping the second one only. I'm not aware of a scenario where the `ensure!` would succeed but the `.get()` wouldn't

Reply:
Thats a good catch - Error::<T>::FailToRetrieveVamm will never be thrown if it reaches that point

Reply:
I see.... 
In that case maybe it would make more sense to keep `VammDoesNotExist` but delete the other error variant. (removing the `ensure!`), right?